src/Core/Types/Types/Descriptors/Conventions/INamingConventions.cs

```diff
@@ -23,8 +23,8 @@ public interface INamingConventions
 
         string GetEnumValueDescription(object value);
 
-        string GetDeprecationReason(MemberInfo member);
+        bool IsDeprecated(MemberInfo member, out string reason);
```

Comment:
Breaking changes for anyone implementing a custom naming convention?

Reply:
Yeah we should probably add the `string GetDeprecationReason(MemberInfo member);` method as obsolete and remove it in a later major update.

Reply:
https://github.com/ChilliCream/hotchocolate/issues/879